libs/blocks/merch-card/merch-card.js

```diff
@@ -1,161 +1,155 @@
-  const [ribbonBgColor, ribbonTextColor] = ribbonStyle.split(', ');
+  if (!ribbonStyleRegex.test(ribbonMetadata[0]?.innerText)) return null;
+  const style = ribbonMetadata[0].innerText;
+  const badgeBackgroundColor = style.split(',')[0].trim();
+  const badgeColor = style.split(',')[1].trim();
   const ribbonWrapper = ribbonMetadata[0].parentNode;
-  const ribbon = ribbonMetadata[1];
-  ribbon.classList.add(`consonant-${cardType}-ribbon`);
-  const borderStyle = `1px solid ${ribbonBgColor}`;
-  if (el.classList.contains('evergreen')) {
-    ribbon.style.border = borderStyle;
-    ribbon.style.borderRight = 'none';
-  } else {
-    ribbon.style.backgroundColor = ribbonBgColor;
-    el.style.border = borderStyle;
-  }
-  ribbon.style.color = ribbonTextColor;
-  const picture = el.querySelector(`.consonant-${cardType}-img`);
-  if (picture) {
-    picture.insertAdjacentElement('afterend', ribbon);
-  } else {
-    el.insertAdjacentElement('beforeend', ribbon);
-  }
+  const badgeText = ribbonMetadata[1].innerText;
   ribbonWrapper.remove();
+  return { badgeBackgroundColor, badgeColor, badgeText };
 };
 
-const decorateIcon = (el, icons, cardType) => {
-  if (!icons) return;
-  const inner = el.querySelector(`.consonant-${cardType}-inner`);
-  const iconWrapper = createTag('div', { class: `consonant-${cardType}-iconWrapper` });
-  icons?.forEach((icon) => {
-    const url = icon.querySelector('img').src;
-    const iconDiv = createTag('div', { class: 'consonant-MerchCard-ProductIcon', style: `background-image: url(${url})` });
-    iconWrapper.appendChild(iconDiv);
-    icon.parentNode?.remove();
-  });
-  inner.prepend(iconWrapper);
+const getActionMenuContent = (el, ribbonMetadata) => {
+  const index = (ribbonMetadata !== null) ? 1 : 0;
+  const expectedChildren = (ribbonMetadata !== null) ? 3 : 2;
+  if (el.childElementCount !== expectedChildren) {
+    return null;
+  }
+  const actionMenuContentWrapper = el.children[index];
+  const actionMenuContent = actionMenuContentWrapper.children[0];
+  actionMenuContentWrapper.remove();
+  return actionMenuContent;
 };
 
+function getMerchCardRows(rows, ribbonMetadata, cardType, actionMenuContent) {
+  if (cardType === 'catalog') {
+    const index = ribbonMetadata === null ? 0 : 1;
+    return actionMenuContent !== null ? rows[index + 1] : rows[index];
+  }
+  return rows[ribbonMetadata === null ? 0 : 1];
+}
+
 const init = (el) => {
-  loadStyle(`${base}/deps/caas.css`);
-  const section = el.closest('.section');
-  section.classList.add('milo-card-section');
+  let section = el.closest('.section');
+  if (section.parentElement.classList.contains('fragment')) {
+    section.style.display = 'contents';
+    const fragment = section.parentElement;
+    fragment.style.display = 'contents';
+    const fragmentParent = fragment.parentElement;
+    fragmentParent.style.display = 'contents';
+    const parentSection = fragmentParent.parentElement;
+    section = parentSection;
```

Comment:
Might be cleaner if you put all the `const`s at top, and all the style modifications below. Also, not sure why you reassign section on 136, I don't see it being used further in this file.